gcore/tilematrixset.cpp

```diff
@@ -77,7 +77,8 @@ std::unique_ptr<TileMatrixSet> TileMatrixSet::parse(const char *fileOrDef)
     std::unique_ptr<TileMatrixSet> poTMS(new TileMatrixSet());
 
     constexpr double HALF_CIRCUMFERENCE = 6378137 * M_PI;
-    if (EQUAL(fileOrDef, "GoogleMapsCompatible"))
+    if (EQUAL(fileOrDef, "http://www.opengis.net/def/tilematrixset/OGC/1.0/GoogleCRS84Quad")
```

Comment:
the old names must still be supported, otherwise it breaks other things in GDAL. Cf https://app.travis-ci.com/github/OSGeo/gdal/jobs/601224478
```
________________________ ERROR collecting gcore/cog.py _________________________

gcore/cog.py:302: in <module>
    @gdaltest.require_creation_option("COG", "JPEG")
pymod/gdaltest.py:1590: in decorator
    drv is None or option not in drv.GetMetadata()["DMD_CREATIONOPTIONLIST"],
/usr/lib/python3/dist-packages/osgeo/gdal.py:2818: in GetMetadata
    return self.GetMetadata_Dict(domain)
/usr/lib/python3/dist-packages/osgeo/gdal.py:2789: in GetMetadata_Dict
    return _gdal.MajorObject_GetMetadata_Dict(self, *args)
E   RuntimeError: Invalid tiling matrix set name
```